service/controller/v17/resource/ipam/create.go

```diff
@@ -0,0 +1,167 @@
+package ipam
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"reflect"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/ipam"
+	"github.com/giantswarm/microerror"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/aws-operator/service/controller/v17/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/v17/key"
+)
+
+// EnsureCreated allocates guest cluster network segment. It gathers existing
+// subnets from existing AWSConfig/Status objects and existing VPCs from AWS.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "checking if subnet needs to be allocated for cluster")
+
+	if key.ClusterNetworkCIDR(customObject) == "" {
+		var subnetCIDR string
+		// TODO(tuommaki): Remove this when all tenant clusters are upgraded to
```

Comment:
TODO issue here: https://github.com/giantswarm/giantswarm/issues/4192

Reply:
Please link the issue in the comment.

Reply:
Sure.